src/openms/source/ANALYSIS/ID/ConsensusIDAlgorithmPEPIons.cpp

```diff
@@ -109,29 +109,27 @@ namespace OpenMS
     set<double> matches; // each best-matching ion counts only once
     vector<double>::iterator start = ions2.begin();
     // for each fragment in seq. 1...
-    for (vector<double>::iterator it1 = ions1.begin(); it1 != ions1.end();
-         ++it1)
+    for (auto& ion1 : ions1)
     {
       // ...find fragments from seq. 2 that are within the mass tolerance:
       vector<double>::iterator lower = lower_bound(start, ions2.end(),
-                                                   *it1 - mass_tolerance_);
+                                                   ion1 - mass_tolerance_);
       if (lower == ions2.end()) break; // all values are too low
-      vector<double>::iterator upper = upper_bound(lower, ions2.end(),
-                                                   *it1 + mass_tolerance_);
+
```

Comment:
what happened to upper_bound?

Reply:
Hmm, that looks like it got mangled in translation. I'll revert this